    /** Do 1.
     * @throws TestException1 when error occurrs.
     */
    public void doStuff1() throws TestException1 {
        try {
            doStuff2();
        } catch (TestException2 e) { }
        throw new Test1().new TestException1("");
    }

